### PR TITLE
Species Specific Agent Items

### PR DESCRIPTION
**Pull request by MarsM0nd**
WIP for species specific agent items, similar to job specific agent items. Still got to get some sprites for the current implementation, and probably should also add more than just one agent item. The way it's implemented might also not be the best, not sure, hence why i would like an early look over this code.

### CODE REVIEW

**Comment by Citinited** [comment]
Sure, why not? This might make traitors of different races feel a little more differentiated.

**Comment by FalseIncarnate** [comment]
I'd actually argue these should be a bit more than citinited said, since in addition to the damage, these new claws are also effectively half a loaded toolbox worth of tools. Also not sure about giving these to Kidan, but I guess the ants do need something.

Definitely a neat idea, interested to see what other species items might be next. Perhaps slime ventcrawl might return as one?

**Comment by jughu** [comment]
Only if vox can buy voxleap for 20 tc

**Comment by MarsM0nd** [comment]
Yeah, the TCs are also still a work in progress, i would think of 5 tc for the claws. I did have the intention with this that the racial powers come back with that, including voxleap, for a good price, and slime ventcrawling. This was made as a proof of concept at first, so i did want this to be checked codewise, did make a corresponsive thread on the forums: https://nanotrasen.se/forum/topic/9911-species-specific-agent-items/

**Comment by taukausanake** [comment]
Could you explain why you have wirecutter, screwdriver, and of all things crowbar claws? Are these functional tools as well as things you can use to cut people up?

**Comment by MarsM0nd** [comment]
Yeah, working as the toolset, good enough to get through a door if you know the wires or a window, and edagger dammage. I did give them crowbar claws, as i wanted to give them more than just a nodrop edagger, cutting with the claws should be self-explanatory, screwdiving should be realistic with hardened claws, do have a thin edge, so you can use that to turn a screw, and the crowbar would be some combination of digging the claws into the object, and using enhanced muscles instead of a lever to move something. This way, you have the small, but functional, toolset. The tool-speed values probably need adjusting, again, still the proof of concept implementation.

**Comment by uraniummeltdown** [comment]
As long as species are karma-locked :-1:

**Comment by Fox-McCloud** [comment]
Pretty much what @uraniummeltdown said.

Even then, I'm not really too keen on this. The purpose of traitor items are to offer near universal experience to as many crew as possible. Admittedly, job specific traitor items *do* skirt this a little bit, but for the most part, are generally more gimmicky (or specialized) equivalents of wider traitor items.

Jobs are generally available equally, and are unchangeable at round start; species? Not so much.

**Comment by Regen1** [comment]
This seems more like a power creep than anything, 1 more TC for something that is completely hidden, doesn't take up an inventory space and supposedly works as multiple tools?

This is super OP if my half asleep brain can read it correctly.

File: code/game/objects/items/weapons/species_enhancement/species_enhancement.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		icon_state = "lepopen"
+
+/obj/item/weapon/species_enhancement/attack(mob/M as mob, mob/user as mob)
+	if(!M || !user)
+		return
+
+	if(!ishuman(M) || !ishuman(user))
+		return
+
+	if(src.used)
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/species_enhancement/species_enhancement.dm]: `src.` is bad, it's implied already on any variables

File: code/game/objects/items/weapons/species_enhancement/species_enhancement.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(species && species.len) //Species specific enhancer without a species restriction could be adminbused, hence sanity check
+			if(!(species.Find(H.species.name)))
+				to_chat(user, "<span class='warning'>You failed to inject [M], as they are incompatible with this genetic modifier.</span>")
+				return
+
+	if(M == user)
+		user.visible_message("<span class='danger'>[user] injects \himself with [src]!</span>")
+		src.injected(user)
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/species_enhancement/species_enhancement.dm]: :eyes:

File: code/game/objects/items/weapons/species_enhancement/species_enhancement.dm [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	else
+		user.visible_message("<span class='danger'>[user] is trying to inject [M] with [src]!</span>")
+		if(do_mob(user,M,30))
+			user.visible_message("<span class='danger'>[user] injects [M] with [src].</span>")
+			src.injected(M)
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/species_enhancement/species_enhancement.dm]: :eyes:

File: code/game/objects/items/weapons/species_enhancement/species_enhancement.dm [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		else
+			to_chat(user, "<span class='warning'>You failed to inject [M].</span>")
+
+/obj/item/weapon/species_enhancement/proc/injected(var/mob/living/carbon/human/target)
+	src.used = 1
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/species_enhancement/species_enhancement.dm]: :eyes:

File: code/game/objects/items/weapons/species_enhancement/species_enhancement.dm [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	src.update_icon()
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/species_enhancement/species_enhancement.dm]: :eyes:

File: code/game/objects/items/weapons/species_enhancement/species_enhancement.dm [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	src.name = "used " + src.name
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/species_enhancement/species_enhancement.dm]: :eyes:

File: code/game/objects/items/weapons/species_enhancement/species_enhancement.dm [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+	if(!M || !user)
+		return
+
+	if(!ishuman(M) || !ishuman(user))
+		return
+
+	if(src.used)
+		return
+
+	if(ishuman(M))
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/species_enhancement/species_enhancement.dm]: kill this whole block off and replace it with `if(!species.Find(M.get_species()))`

File: code/game/objects/items/weapons/species_enhancement/species_enhancement.dm [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+	var/used = null
+	var/list/species = null
+
+/obj/item/weapon/species_enhancement/update_icon()
+	if(used)
+		icon_state = "lepopen0"
+	else
+		icon_state = "lepopen"
+
+/obj/item/weapon/species_enhancement/attack(mob/M as mob, mob/user as mob)
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/species_enhancement/species_enhancement.dm]: drop the `as mob`, they do nothing

File: code/game/objects/items/weapons/species_enhancement/special_claws.dm [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+	qdel(src)
+	user.put_in_active_hand(new_claws)
+
+/obj/item/weapon/crowbar/claws/dropped()
+	..()
+	loc = null
+
+/obj/item/weapon/screwdriver/claws
+	name = "Claws"
+	desc = "Your claws, currently ready to be stuck into something."
```
Comment (KingPhilipIII) [inline code/game/objects/items/weapons/species_enhancement/special_claws.dm]: How lewd.

File: code/datums/uplink_item.dm [hunk 10]
```diff
@@ -358,6 +359,18 @@ var/list/uplink_items = list()
...
+//Work in Progress, species specific antag tools
+/datum/uplink_item/speciesspecific
+	category = "Species Specific Tools"
+
+/datum/uplink_item/speciesspecific/claw_sharpener
+	name = "Claw Sharpener"
+	desc = "With some genetic and physicial modifications, claws can become very deadly and useful"
+	item = /obj/item/weapon/species_enhancement/claws
+	cost = 3
+	species = list("Tajaran", "Unathi", "Vulpkanin", "Kidan")
```
Comment (KasparoVy) [inline code/datums/uplink_item.dm]: Might be best to do away with this, and...

File: code/game/objects/items/devices/uplinks.dm [hunk 11]
```diff
@@ -69,6 +74,9 @@ var/list/world_uplinks = list()
 			if(I.job && I.job.len)
 				if(!(I.job.Find(job)))
 					continue
+			if(I.species && I.species.len)
+				if(!(I.species.Find(species)))
```
Comment (KasparoVy) [inline code/game/objects/items/devices/uplinks.dm]: have this be iterated on per-object basis so you can check for species with claws by flag instead

**Comment by fludd12** [comment]
With how the claws work as not only a fairly high damage weapon, but half a tool belt, and concealable at will, I think they should be something like 5-10 TC. Maybe I'm just a debuff nerd, though.

**Comment by Lady-Luck** [comment]
You could technically make it applicable for all races and have it be a toggle.

**Comment by MarsM0nd** [comment]
Did make some balancing, added the Vox to include every species with claws, and the better code Tiger did comment on.
I understand the concern that some people might never really get to toy around with certain agent equipment, and that with changelings, it might cause a person to have something that doesn't fit their species. But this does try to make species more different from another, and in a way, you could see the problem about beeing locked out of a certain mechanic with the karma races already, as slimes for example can regrow limbs, don't need robotics or surgery.
Another idea would be making the species themselves handle some sort of "genetic overdrive", which would give them certain verbs if activated, which would make this universal again, and would also allow changeling to get that.

File: code/datums/uplink_item.dm [hunk 12]
```diff
@@ -358,6 +359,18 @@ var/list/uplink_items = list()
...
 
 
+
+//Work in Progress, species specific antag tools
+/datum/uplink_item/speciesspecific
+	category = "Species Specific Tools"
+
+/datum/uplink_item/speciesspecific/claw_sharpener
+	name = "Claw Sharpener"
+	desc = "With some genetic and physicial modifications, claws can become very deadly and useful"
```
Comment (SamHPurp) [inline code/datums/uplink_item.dm]: Full stop or other sentence ender.

**Comment by Regen1** [comment]
Okay, so, let me get these claws straight

They do the same damage, does not take up a slot in your inventory and / or hands, has no activation sound, has no sprite in hand, is completely impossible to distinguish from normal claws unless security forces you to attack someone and they look at how much damage you did, functions as 3 tools in one, being screwdriver, crowbar and wirecutter (this also mean you can ghetto hack any airlock without any tools), can't be disarmed or lost.

I'm just going to say that you can throw the idea away, this is something to incredibly OP that it is in line with guardian spirits, as it gives you such a plethora of tools and a weapon which can't be removed from you unless you're killed and cloned.

**Comment by jughu** [comment]
i mean if it was 20 tc i wouldnt mind it tbh if you can atleast surgically remove them or come up on surgery sensor thingies

**Comment by MarsM0nd** [comment]
Stripped this down to the framework, can't really focus as much as i did want to on it, and it wasn't really clear from @Fox-McCloud comment if species specific tools like this is wanted. This can be closed if it's not wanted, merged to allow others to come up with species specific items and implementations, the claws that i did weren't really that well thought out, just made them so i can test the code myself.

**Comment by Fox-McCloud** [comment]
Going to turn this down. Don't think it's a good idea to be tying what items are available to you to your species; that should be something that's, at worst, tied to your job--species specifics is a whole can of worms that I don't think is going to add *good* depth to SS13.

**Pull request closed without merging**